Revert DAML-LF proto change for Numeric (#2600)
digital-asset/daml: unreleased.rst

  the performance of ``damlc`` and ``daml studio`` on Windows.
+ [DAML Compiler] ``damlc build`` should no longer leak file handles so
  ``ulimit`` workarounds should no longer be necessary.

> + [DAML-LF] **Breaking** Rename back ``NUMERIC`` by ``DECIMAL`` in archive Protobuf definition.

Reviewer: ```suggestion
+ [DAML-LF] **Breaking** Rename ``NUMERIC`` back to ``DECIMAL`` in Protobuf definition.
```
